test/integration/read_calls_strategies_test.lua

```diff
@@ -0,0 +1,155 @@
+local fio = require('fio')
+
+local t = require('luatest')
+
+local helpers = require('test.helper')
+
+local pgroup = helpers.pgroup.new('read_calls_strategies', {
+    engine = {'memtx'},
+})
+
+pgroup:set_before_all(function(g)
+    g.cluster = helpers.Cluster:new({
+        datadir = fio.tempdir(),
+        server_command = helpers.entrypoint('srv_simple_operations'),
+        use_vshard = true,
+        replicasets = helpers.get_test_replicasets(),
+        env = {
+            ['ENGINE'] = g.params.engine,
+        },
+    })
+
+    g.cluster:start()
+
+    g.space_format = g.cluster.servers[2].net_box.space.customers:format()
+
+    -- wrap call_impl and call_single_impl
+    local call_impl_funcs = {'call_impl', 'call_single_impl'}
+    g.cluster.main_server.net_box:eval([[
+        local call = require('crud.common.call')
+        call.internal.vshard_calls = {}
+
+        local call_impl_funcs = ...
+        for _, func_name in ipairs(call_impl_funcs) do
+            call.internal.vshard_calls[func_name] = {}
+
+            local current_func = call.internal[func_name]
+            call.internal[func_name] = function(...)
+                local vshard_call = ...
+                table.insert(call.internal.vshard_calls[func_name], vshard_call)
+                return current_func(...)
+            end
+
```

Comment:
Maybe you can use `box.stat().CALL.total` for such purpose. I don't like approach with internals monkeypatching

Reply:
I've monkeypatched `router_mt`.